src/main/java/io/anserini/index/generator/HnswJsonWithSafeTensorsDenseVectorDocumentGenerator.java

```diff
@@ -0,0 +1,166 @@
+package io.anserini.index.generator;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.anserini.collection.SourceDocument;
+import io.anserini.index.Constants;
+import org.apache.lucene.document.BinaryDocValuesField;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.KnnFloatVectorField;
+import org.apache.lucene.document.StringField;
+import org.apache.lucene.index.VectorSimilarityFunction;
+import org.apache.lucene.util.BytesRef;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Converts a {@link SourceDocument} from SafeTensors into a Lucene {@link Document}, ready to be indexed.
+ *
+ * @param <T> type of the source document
+ */
+public class HnswJsonWithSafeTensorsDenseVectorDocumentGenerator<T extends SourceDocument> implements LuceneDocumentGenerator<T> {
+  
+
+
+    public HnswJsonWithSafeTensorsDenseVectorDocumentGenerator() {
+       
+    }
+
+    // public void setCurrentJsonlFile(String currentJsonlFile) {
+    //     this.currentJsonlFile = currentJsonlFile;
+    // }
+
+    @Override
+    public Document createDocument(T src) throws InvalidDocumentException {
+        try {
+
+            String vectorsFilePath = "collections/beir-v1.0.0/bge-base-en-v1.5.safetensors/nfcorpus/vectors.part00_vectors.safetensors";
+            String docidsFilePath = "collections/beir-v1.0.0/bge-base-en-v1.5.safetensors/nfcorpus/vectors.part00_docids.safetensors";
+            String docidToIdxFilePath = "collections/beir-v1.0.0/bge-base-en-v1.5.safetensors/nfcorpus/vectors.part00_docid_to_idx.json";
```

Comment:
there shouldn't be any more json?

Reply:
Since the vector IDs are strings, we created a dictionary to map these IDs to indices, allowing us to reference them efficiently when we deserialize the `docids.safetensors.`

Reply:
but why JSON?